### DISCUSSION TITLE
uninstall  WARNING: Any MPI toolboxes that were installed with the help of blueCFD-Core 2020, were not uninstalled automatically.y

### DISCUSSION THREAD

**Discussion by avcortiz** [blueCFD/Core#209] [Q&A] [answered]
![error](https://user-images.githubusercontent.com/82894737/115426901-f1620580-a1ce-11eb-92b9-f4f35d374594.JPG)
I do not know what to do next

**Comment by wyldckat** [comment, marked as answer]
Sorry about that, it's one of those situations where it made more sense in our head during development...

The idea is that if the provided MS-MPI toolbox was installed during blueCFD-Core's installation, then our uninstaller will not take care of uninstalling said MS-MPI toolboxes. There are a couple of reasons why we chose to do this:

1. Even if the option was chosen in blueCFD-Core's installer, namely to also install the provided MS-MPI on Windows, it doesn't mean that it really was installed or not, since it allows the end-user to choose to not install it.

2. Even though when blueCFD-Core installer tries to install MS-MPI, it might already be installed, therefore we can't uninstall it automatically with blueCFD-Core's uninstaller.

3. It is also possible that there was a more recent MS-MPI toolbox already installed, in which case there isn't much we can do.

So, to answer your question:
* Since it's just a warning, you can ignore this warning.
* It only becomes a problem if by any chance the MS-MPI version that was installed by blueCFD-Core has messed up something in Windows and is as of now giving problems with other applications. But we are not aware of such a situation.


edit: Please also check the discussion #204, which has additional instructions on how to use an older MSMPI version, if a newer one is already installed in Windows.

**Reply by tahirzahid995** [reply]
I had the same issue. I had installed then uninstalled blueCFD-Core 2020 and got the same message. For working in butterfly(in grasshopper), I installed blueCFD-Core 2017-2 but the MP drivers got bypassed since a newer version was already installed. However, the simulation isn't running correctly so there must be something wrong with them. I want to find the MP drivers and uninstall them manually so I can reinstall blueCFD-Core 2017-2 from scratch. Can you provide advice?

**Reply by wyldckat** [reply]
@tahirzahid995: In discussion #204, see solution 1: https://github.com/blueCFD/Core/discussions/204#solution-1
* The idea is that you have to start the blueCFD-Core 2017-2 terminal and then you should be able to run the commands given in that solution.

The other possibility is to remove the MSMPI version that was installed in Windows, which can be done by following the instructions given here: https://github.com/blueCFD/Core/discussions/205#discussioncomment-2310520

**Reply by tahirzahid995** [reply]
Thank you ! Much appreciated.
